packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -427,6 +444,37 @@ const MetricForm: FC<MetricFormProps> = ({
     }
   }, [type, form]);
 
+  const tableOptions: { label: string; value: string }[] = [];
+
+  const { data: TableData } = useApi<{
+    informationSchema: InformationSchemaInterface;
+  }>(`/datasource/${selectedDataSource.id}/schema`);
+
+  if (TableData?.informationSchema?.databases.length) {
+    TableData.informationSchema.databases.forEach((database) => {
+      database.schemas.forEach((schema) => {
+        schema.tables.forEach((table) => {
+          tableOptions.push({ label: table.tableName, value: table.id });
+        });
+      });
+    });
+  }
+
+  const columnOptions: { label: string; value: string }[] = [];
+
+  const { data: columnData } = useApi<{
+    table: InformationSchemaTablesInterface;
+  }>(`/datasource/${selectedDataSource.id}/schema/table/${tableId}`);
+
+  if (columnData?.table?.columns.length) {
+    columnData.table.columns.forEach((column) => {
+      columnOptions.push({
+        label: column.columnName,
+        value: column.columnName,
+      });
+    });
+  }
```

Comment:
All of this logic seems like it could be in a custom hook. We could use it any time we want to ask the user to select a table/column.

```ts
const {
  table,
  setTable,
  tableOptions,
  columnOptions,
} = useSchemaFormOptions(datasourceId);
```

Reply:
Done!